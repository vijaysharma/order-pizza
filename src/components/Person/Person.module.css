.Person {
  display: flex;
  align-items: center;
  justify-content: center;
  padding: 2rem 1rem 1rem;
  border-top: 2px solid #8b8b8b;
  margin-top: 1rem;
}
.Person:last-child {
  margin-top: 0;
  padding-bottom: 0;
}
.Person > span {
  font-size: 2rem;
  width: 3rem;
}

.Person > h2 {
  font-size: 1.5rem;
  font-weight: 400;
  margin: 0;
  flex-grow: 1;
}

.Person > div {
  text-align: right;
  flex-grow: 1;
}