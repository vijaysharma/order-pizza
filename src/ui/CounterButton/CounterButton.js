import React from 'react';
import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
import {faMinusCircle, faPlusCircle} from '@fortawesome/free-solid-svg-icons';
import * as classes from './CounterButton.module.css';

const CounterButton = props => (
  <div className={classes.CounterButton}>
    <FontAwesomeIcon icon={faMinusCircle} onClick={props.decrement} className={[props.disabled ? classes.DecrementDisabled : '', classes.Decrement].join(' ')} />
    <span>{props.count}</span>
    <FontAwesomeIcon icon={faPlusCircle} onClick={props.increment} className={classes.Increment} />
  </div>
);


export default CounterButton;