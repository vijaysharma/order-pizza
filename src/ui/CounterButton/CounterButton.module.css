.CounterButton {
  color: #9b9b9f;
  display: inline-block;
}
.CounterButton svg {
  vertical-align: middle;
}
.CounterButton span {
  font-size: 1rem;
  padding: 0 0.75rem;
  margin: 0 0.75rem;
  display: inline-block;
  vertical-align: middle;
  width: 2.5rem;
  text-align: center;
}
.Increment, .Decrement {
  cursor: pointer;
  font-size: 2rem;
}
.Increment {
  color: #e13b5b;
}
.Decrement {
  color: #283688;
}
.DecrementDisabled {
  color: #8b8b8b;
  cursor: not-allowed;
}