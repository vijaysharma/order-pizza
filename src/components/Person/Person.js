import React from 'react';
import CounterButton from '../../ui/CounterButton/CounterButton';
import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
import {faChild, faUser} from '@fortawesome/free-solid-svg-icons';
import * as classes from './Person.module.css';

const Person = props => {
  const label = props.label ? props.label.toUpperCase() : '';
  return (
    <div className={classes.Person}>
      <span><FontAwesomeIcon icon={props.label === 'adults' ? faUser : faChild} /></span>
      <h2>{label}</h2>
      <div>
        <CounterButton
          count={props.count}
          increment={props.onIncrement}
          decrement={props.onDecrement}
          disabled={props.disabled} />
      </div>
    </div>
  );
};

export default Person;