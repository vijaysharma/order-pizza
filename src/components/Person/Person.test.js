import React from 'react';

import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Person from './Person';
import CounterButton from '../../ui/CounterButton/CounterButton';
import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';

configure({adapter: new Adapter()});

describe('<Person/>', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Person />);
  });

  it('should render <PizzaSlice/> component', () => {
    expect(wrapper.find(CounterButton)).toHaveLength(1);
  });

  it('should render <PizzaSlice/> correctly', () => {
    wrapper.setProps({
      label: 'adults',
      count: 1
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('should render <PizzaSlice/> component with label ADULTS', () => {
    wrapper.setProps({
      label: 'adults',
      count: 1
    });
    expect(wrapper.find('h2').text().includes('ADULTS')).toBe(true);
  });

  it('should render <PizzaSlice/> component with icon', () => {
    expect(wrapper.find(FontAwesomeIcon)).toHaveLength(1);
  });


});