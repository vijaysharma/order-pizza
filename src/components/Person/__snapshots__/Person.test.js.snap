// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Person/> should render <PizzaSlice/> correctly 1`] = `ShallowWrapper {}`;
